src/librbd/managed_lock/GetLockerRequest.cc

```diff
@@ -103,14 +103,14 @@ void GetLockerRequest<I>::handle_get_lockers(int r) {
     return;
   }
 
-  m_locker->entity = iter->first.locker;
-  m_locker->cookie = iter->first.cookie;
-  m_locker->address = iter->second.addr.get_legacy_str();
-  if (m_locker->cookie.empty() || m_locker->address.empty()) {
+  if (iter->first.cookie.empty() || iter->second.addr.is_blank_ip()) {
```

Comment:
I see two issues:
- `iter->first.cookie.empty()` condition is dead because `util::decode_lock_cookie()` above would treat any string that isn't of `auto <number>` format as an external cookie, causing us to bail out with "locked by external mechanism" error
- I don't see how `iter->second.addr.is_blank_ip()` can arise under normal conditions.  The only reason you ran into it is some missing piece in Crimson, right?

Treating a locker that has a valid cookie but a blank address as if there were no lockers seems wrong.  I would suggest changing this piece of code as follows:
```
  if (!util::decode_lock_cookie(iter->first.cookie, &m_locker->handle)) {
    ldout(m_cct, 5) << "locked by external mechanism: "
		    << "cookie=" << iter->first.cookie << dendl;
    finish(-EBUSY);
    return;
  }
  if (iter->second.addr.is_blank_ip()) {
    ldout(m_cct, 5) << "locker has a blank address" << dendl;
    finish(-EBUSY);
    return;
  }

  m_locker->entity = iter->first.locker;
  m_locker->cookie = iter->first.cookie;
  m_locker->address = iter->second.addr.get_legacy_str();
```